libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -379,27 +379,26 @@ string YulUtilFunctions::roundUpFunction()
 	});
 }
 
-string YulUtilFunctions::overflowCheckedUIntAddFunction(size_t _bits)
+string YulUtilFunctions::overflowCheckedIntAddFunction(IntegerType const& _type)
 {
-	solAssert(0 < _bits && _bits <= 256 && _bits % 8 == 0, "");
-	string functionName = "checked_add_uint_" + to_string(_bits);
+	string functionName = "checked_add_" + _type.identifier();
 	return m_functionCollector->createFunction(functionName, [&]() {
 		return
 			Whiskers(R"(
 			function <functionName>(x, y) -> sum {
-				<?shortType>
```

Comment:
So the `shortType` stuff wasn't really needed?

Reply:
It was needed in the old version (that checked for overflow *after* adding), but is no longer *needed* for the new version (that checks for overflow *before* adding).

For short types the  new version is actually cheaper - but I'm just realizing that for ``uint256`` the old version was in fact slightly cheaper, so maybe it makes sense to keep the special case for that...

Reply:
The ``mul`` case is different - there it was entirely superfluous...

Reply:
But yeah - I think I'll change the unsigned non-short-type case back to the old version (reintroducing the case distinction) - adding two ``uint256``s is probably among the most common operations, so I guess it makes sense to sacrifice some brevity here for cheaper runtime costs...
Although... if we add a new optimizer rule that changes ``sub(2^256-1, x)`` to ``not(x)`` (which we should probably have anyways, if it's in fact correct), then it'd only be 3 gas difference... I'll think about it tomorrow...

Reply:
Coolest thing would be, if the optimizer recognized, that it can move the check across the ``add`` and save 3 gas by it :-D. But that's probably utopian :-).

Reply:
Added a comment above to reconsider this against real world examples at a later point. https://github.com/ethereum/solidity/pull/6935 will reduce the gas difference to 3 gas and at the moment it's unclear whether making it easier for the optimizer to move up the revert condition is more beneficial (or in fact harmful) in contrast to "pre-optimizing" the 256-bit case here, so for the time being I'd opt for simpler code here.